govcd/vm_test.go

```diff
@@ -1296,3 +1296,186 @@ func deleteVapp(vcd *TestVCD, name string) error {
 	}
 	return nil
 }
+
+func (vcd *TestVCD) Test_AddNewEmptyVMMultiNIC(check *C) {
+
+	config := vcd.config
+	if config.VCD.Network.Net1 == "" {
+		check.Skip("Skipping test because no network was given")
+	}
+
+	// Find VApp
+	if vcd.vapp.VApp == nil {
+		check.Skip("skipping test because no vApp is found")
+	}
+
+	vapp, err := createVappForTest(vcd, "Test_AddNewEmptyVMMultiNIC")
+	check.Assert(err, IsNil)
+	check.Assert(vapp, NotNil)
+
+	desiredNetConfig := &types.NetworkConnectionSection{}
+	desiredNetConfig.PrimaryNetworkConnectionIndex = 0
+	desiredNetConfig.NetworkConnection = append(desiredNetConfig.NetworkConnection,
+		&types.NetworkConnection{
+			IsConnected:             true,
+			IPAddressAllocationMode: types.IPAllocationModePool,
+			Network:                 config.VCD.Network.Net1,
+			NetworkConnectionIndex:  0,
+		},
+		&types.NetworkConnection{
+			IsConnected:             true,
+			IPAddressAllocationMode: types.IPAllocationModeNone,
+			Network:                 types.NoneNetwork,
+			NetworkConnectionIndex:  1,
+		})
+
+	// Test with two different networks if we have them
+	if config.VCD.Network.Net2 != "" {
+		// Attach second vdc network to vApp
+		vdcNetwork2, err := vcd.vdc.GetOrgVdcNetworkByName(vcd.config.VCD.Network.Net2, false)
+		check.Assert(err, IsNil)
+		_, err = vapp.AddOrgNetwork(&VappNetworkSettings{}, vdcNetwork2.OrgVDCNetwork, false)
+		check.Assert(err, IsNil)
+
+		desiredNetConfig.NetworkConnection = append(desiredNetConfig.NetworkConnection,
+			&types.NetworkConnection{
+				IsConnected:             true,
+				IPAddressAllocationMode: types.IPAllocationModePool,
+				Network:                 config.VCD.Network.Net2,
+				NetworkConnectionIndex:  2,
+			},
+		)
+	} else {
+		fmt.Println("Skipping adding another vdc network as network2 was not specified")
+	}
+
+	cat, err := vcd.org.GetCatalogByName(vcd.config.VCD.Catalog.Name, true)
+	check.Assert(err, IsNil)
+	check.Assert(cat, NotNil)
+
+	//media, err := cat.GetMediaByName("photon-custom-hw11-2.0-304b817.ova", false)
+	media, err := cat.GetMediaByName("vaido2", false)
```

Comment:
hardcoded name. How does this media item become available?

Reply:
updated. Value comes from configuration